yt/visualization/_handlers.py

```diff
@@ -0,0 +1,458 @@
+        ):
+            dvmax = self.to_float(self.vmax)
+        elif np.any(finite_values_mask):
+            dvmax = self.to_float(np.nanmax(data[finite_values_mask]))
+
+        if self.dynamic_range is not None:
+            dvmin, dvmax = self.get_dynamic_range(dvmin, dvmax)
+
+        if dvmin is None:
+            dvmin = 1 * getattr(data, "units", 1)
+        kw.setdefault("vmin", dvmin)
+
+        if dvmax is None:
+            dvmax = 1 * getattr(data, "units", 1)
+        kw.setdefault("vmax", dvmax)
+
+        min_abs_val, max_abs_val = np.sort(np.abs((kw["vmin"], kw["vmax"])))
+        if self.norm_type is not None:
+            # this is a convenience mechanism for backward compat,
+            # allowing to toggle between lin and log scaling without detailed user input
+            norm_type = self.norm_type
+        else:
+            if kw["vmin"] == kw["vmax"] or not np.any(finite_values_mask):
+                norm_type = Normalize
+            elif kw["vmin"] <= 0:
+                norm_type = SymLogNorm
+            else:
+                norm_type = LogNorm
+
+        if norm_type is SymLogNorm:
+            if self.linthresh is not None:
+                linthresh = self.to_float(self.linthresh)
+            else:
+                linthresh = self.to_float(self._guess_linthresh(data))
+
+            kw.setdefault("linthresh", linthresh)
+            if MPL_VERSION >= Version("3.2"):
+                # note that this creates an inconsistency between mpl versions
+                # since the default value previous to mpl 3.4.0 is np.e
+                # but it is only exposed since 3.2.0
+                kw.setdefault("base", 10)
+
+        return norm_type(*args, **kw)
+
+    def _guess_linthresh(self, data):
+        # data is the ImageArray to plot
+
+        # get the data range as unyt quantities
+        nz_data = data[data != 0]
+        min_abs_val, max_abs_val = np.sort(
+            (np.abs(np.nanmin(nz_data)), np.abs(np.nanmax(nz_data)))
+        )
+        min_abs_val = self.ds.quan(min_abs_val, data.units)
+        max_abs_val = self.ds.quan(max_abs_val, data.units)
+        min_abs_nonzero = self.ds.quan(np.nanmin(np.abs(nz_data)), data.units)
+
+        linthresh = min_abs_nonzero
+        # the above works well for most data. But need to control for a few
+        # edge cases related to floating point precision errors.
+
+        # first, set the scaling factor
+        cutoff_sigdigs = 30.0  # max allowable range in significant digits
+        actual_log_range = np.log10(max_abs_val) - np.log10(min_abs_val)
+        if actual_log_range < cutoff_sigdigs:
+            cutoff_sigdigs = np.ceil(actual_log_range)
+
+        if np.log10(max_abs_val) - np.log10(linthresh) > cutoff_sigdigs:
+            # this is required to avoid rounding error in the matplotlib normalization
+            # when the range is too large.
+            linthresh = max_abs_val / (10**cutoff_sigdigs)
+
+        return linthresh
```

Comment:
So this section is the relevant change...

Reply:
This is a smart algorithm.  It seems to match my solution from #3295 but also incorporate the fix for the sigdig error that my solution broke.  Nice way to find a solution!

Reply:
The code snippet rendered above is out of date. The re-written version is the same end result (matching 3295 except in the edge cases), but I re-wrote it to handle the negative and positive ranges sequentially cause it made handling edge cases easier. Just wanted to double check that you saw that!